### PR TITLE
Colour scale in SliceViewer and ColorFill plots should not update on live data changes

### LINKED ISSUE #14818: LiveViewing in ColorFill Plot and SliceViewer
Users want to set the colour scale on these tools, but every time the new data comes through the scales get re-written. Should be a way to permanently set the scales so that re-scaling doesn't happen upon workspace reload.

This sounds like a small annoying problem, but it's actually stopping users using the live data mode at ISIS

Ask Owen for more details.

### PR DESCRIPTION
**Pull request by tom-perkins**
Resolves #14818 

When the SliceViewer or a ColorFill plot is being used to visualise live data, the user's carefully set colour scale should not be reset every time the data is updated. This PR disables the scale reset if the data is live.

The SliceViewer checks if the workspace updated contains live data by looking for currently running instances of the `MonitorLiveData` algorithm, and comparing their output workspace names with the current workspace name. The code creating the color fill plot works in a similar way, passing the 'is live' flag to the constructor of Spectrogram that takes a workspace. In both cases, the colour scale is not reset on each update if the data is live.

**Test:**
- Start a fake live data listener as [here](http://www.mantidproject.org/MBC_Live_Data_Simple_Examples), e.g. ISIS_Histogram.
- Launch the SliceViewer from the workspace containing live data
- Change the colour scale and make sure it doesn't change back when new data comes in
- Launch a colour fill plot from the workspace containing live data
- Change the colour scale and check it remains the same when new data arrives
- Check that this still works when more than one live data listener is running

### CODE REVIEW

File: MantidQt/SliceViewer/src/SliceViewer.cpp [hunk 1]
```diff
@@ -123,6 +123,14 @@ SliceViewer::SliceViewer(QWidget *parent)
   // hide unused buttons
   ui.btnZoom->hide(); // hidden for a long time
 
+  // If live data being read, save the name of the workspace(s)
+  m_liveDataWSNames.clear();
+  auto liveAlgorithms =
+      AlgorithmManager::Instance().runningInstancesOf("MonitorLiveData");
```
Comment (OwenArnold) [inline MantidQt/SliceViewer/src/SliceViewer.cpp]: I'm not sure about this. Won't this change the behaviour of all plots if we are running this algorithm? This should be a workspace dependent thing.

File: MantidQt/SliceViewer/src/SliceViewer.cpp [hunk 2]
```diff
@@ -123,6 +123,14 @@ SliceViewer::SliceViewer(QWidget *parent)
...
+  for (auto liveAlg : liveAlgorithms) {
+    m_liveDataWSNames.push_back(liveAlg->getPropertyValue("OutputWorkspace"));
```
Comment (OwenArnold) [inline MantidQt/SliceViewer/src/SliceViewer.cpp]: Ok. I see that you're just using that to get the names of OutputWorkspaces.

File: MantidPlot/src/Spectrogram.cpp [hunk 3]
```diff
@@ -48,35 +49,26 @@
...
+Spectrogram::Spectrogram(const QString &wsName,
+                         const Mantid::API::IMDWorkspace_const_sptr &workspace,
+                         const bool isLiveData)
+    : QObject(), QwtPlotSpectrogram(), d_matrix(NULL), d_funct(NULL),
+      d_wsData(NULL), d_wsName(), color_axis(QwtPlot::yRight),
+      color_map_policy(Default), mColorMap(), m_liveData(false) {
+  // First time round, always update color scale, whether live data or not.
   d_wsData = dataFromWorkspace(workspace);
+  // Then set "is live data" to correct value
+  m_liveData = isLiveData;
```
Comment (OwenArnold) [inline MantidPlot/src/Spectrogram.cpp]: Can't this be done in an initialiser list?

**Comment by OwenArnold** [comment]
On Reflection. I don't think this is the right way to solve the issue. The problem is more about being able to specify that an axis should not update rather than it being connected to the LiveData mode.

1. There is already a global setting for 2D plots that should stop auto-updates. I assume this is not being applied to the SliceViewer?
1. We need a way to right-click on our axes to specify that auto-updates should be disabled.
1. The sliceviewer should also have a checkbox for this.

**Comment by tom-perkins** [comment]
Following discussion above, another PR has been opened to solve this issue in a different way (#14909). Closing this PR.

**Pull request closed without merging**